File: software/examples/csharp/ExampleWaterBoiling.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Temperature IR Bricklet 2.0

	// Callback function for object temperature reached callback (parameter has unit °C/10)
	static void ObjectTemperatureCB(BrickletTemperatureIRV2 sender, short temperature)
	{
		Console.WriteLine("Object Temperature: " + temperature/10.0 + " °C");
		Console.WriteLine("The water is boiling!");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletTemperatureIRV2 tir =
		  new BrickletTemperatureIRV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Set emissivity to 0.98 (emissivity of water, 65535 * 0.98 = 64224.299)
		tir.SetEmissivity(64224);

		// Register object temperature reached callback to function ObjectTemperatureCB
		tir.ObjectTemperatureCallback += ObjectTemperatureCB;

		// Configure threshold for object temperature "greater than 100 °C" (unit is °C/10)
		// with a debounce period of 10s (10000ms)
		tir.SetObjectTemperatureCallbackConfiguration(10000, false, '>', 100*10, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleWaterBoiling.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletTemperatureIRV2;

public class ExampleWaterBoiling {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Temperature IR Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletTemperatureIRV2 tir =
		  new BrickletTemperatureIRV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Set emissivity to 0.98 (emissivity of water, 65535 * 0.98 = 64224.299)
		tir.setEmissivity(64224);

		// Add object temperature reached listener (parameter has unit °C/10)
		tir.addObjectTemperatureListener(new BrickletTemperatureIRV2.ObjectTemperatureListener() {
			public void objectTemperature(int temperature) {
				System.out.println("Object Temperature: " + temperature/10.0 + " °C");
				System.out.println("The water is boiling!");
			}
		});

		// Configure threshold for object temperature "greater than 100 °C" (unit is °C/10)
		// with a debounce period of 10s (10000ms)
		tir.setObjectTemperatureCallbackConfiguration(10000, false, '>', 100*10, 0);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
